utils/fwutils.py

```diff
@@ -234,6 +255,139 @@ def get_images(self) -> tuple[str, str]:
+                download_dir = f"{temp_dir}/{self.repo_url}/fixboard/{number}"  # Or any preferred directory for temp storage
+                makedirs(download_dir, exist_ok=True)
+                fixboard_files = list_http_directory(base_url)
+                fixboard_local_file_paths: list[str] = []
+                for file in fixboard_files:
+                    fixboard_local_file_paths.append(
+                        download_file(f"{base_url}/{file}", download_dir)
+                    )
+
+                # Find the required .bin files
+                self.logger.debug(
+                    f"fixboard_local_file_paths: {fixboard_local_file_paths}"
+                )
+                for fixboard_file in fixboard_local_file_paths:
+                    if fixboard_file.endswith(".bin.board_config"):
+
+                        full_address = f"root@{self.imc_address}"
+
+                        file_name = fixboard_file.split("/")[-1]
+                        result = run(
+                            f"scp -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' {fixboard_file} {full_address}:/tmp/{file_name}",
+                            dry_run=self.dry_run,
+                        )
+                        if result.returncode:
+                            self.logger.error(
+                                f"Couldn't transfer file using scp, Error: {result.err}"
+                            )
+                            exit(1)
+                        return f"/tmp/{file_name}"
+
+                self.logger.error("Couldn't find the board_config file, exitting...")
+                exit(1)
+        else:
+            self.logger.error(
+                f"server {self.imc_address} with number {number} doesn't have pre built fixboard images yet, please add the necessary files to the repo"
+            )
+            exit(1)
+
+    def apply_fixboard(self) -> None:
+        fixboard_bin_board_config_file = self.ensure_fixboard_image_on_imc()
+        full_address = f"root@{self.imc_address}"
+        result = ssh_run(
+            "flash_erase /dev/mtd0 0x30000 1",
+            full_address,
+            dry_run=self.dry_run,
+        )
+        if result.returncode:
+            self.logger.error(f"Couldn't flash_erase, Error: {result.err}")
+            exit(1)
+
+        result = ssh_run(
+            f"nandwrite --start=0x30000 --input-size=0x1000 -p /dev/mtd0 {fixboard_bin_board_config_file}",
+            full_address,
+            dry_run=self.dry_run,
+        )
+        if result.returncode:
+            self.logger.error(f"Couldn't nandwrite, Error: {result.err}")
+            exit(1)
+        self.logger.info("Rebooting IMC now!")
+        result = ssh_run(
+            "reboot",
+            full_address,
+            dry_run=self.dry_run,
+        )
+
+    def fixboard_is_needed(self) -> Optional[bool]:
+        full_address = f"root@{self.imc_address}"
+        try:
```

Comment:
Do we expect this to fail?

Reply:
Yes, it can fail because we are interacting with JSON so i just make sure to handle the exception in that function by returning None and wrapping the return type in an Optional